File: treebolicFilesServices/src/main/java/org/treebolic/files/service/TreebolicFilesAIDLBoundService.java
/*
 * Copyright (c) 2023. Bernard Bou
 */

package org.treebolic.files.service;

import org.treebolic.services.TreebolicAIDLBoundService;

import androidx.annotation.NonNull;

/**
 * Bound service for Files data
 */
public class TreebolicFilesAIDLBoundService extends TreebolicAIDLBoundService
{
	/**
	 * Constructor
	 */
	public TreebolicFilesAIDLBoundService()
	{
		super();
	}

	@Override
	public void onCreate()
	{
		super.onCreate();
		this.factory = new ModelFactory(this);
	}

	@SuppressWarnings("SameReturnValue")
	@NonNull
	@Override
	public String getUrlScheme()
	{
		return "directory:";
	}
}

File: treebolicFilesServices/src/main/java/org/treebolic/files/service/TreebolicFilesAIDLBoundService.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.treebolic.files.service

import org.treebolic.services.TreebolicAIDLBoundService

/**
 * Bound service for Files data
 */
class TreebolicFilesAIDLBoundService : TreebolicAIDLBoundService() {

    override fun onCreate() {
        super.onCreate()
        this.factory = ModelFactory(this)
    }

    override val urlScheme: String
        get() = "directory:"
}
